.github/actions/deploy-cloud-run/action.yml

```
@@ -0,0 +1,81 @@
+name: Deploy Cloud Run service
+description: Create a new revision and rollback if health check fails
+inputs:
+  region:
+    description: Region of the Cloud Run service
+    required: true
+  image:
+    description: Image name
+    required: true
+  service:
+    description: Name of the Cloud Run service
+    required: true
+  health_check_url:
+    description: Health check URL. Multiple URLs can be separated by line breaks. If not provided, health check will be skipped.
+
+runs:
+  using: composite
+  steps:
+    - name: Get current revision
+      id: get_revision
+      shell: bash
+      run: |
+        REV=$( \
+          gcloud run services describe "${{ inputs.service }}" \
+            --region "${{ inputs.region }}" \
+            --format="value(status.latestReadyRevisionName)" \
+        )
+        echo "Current revision: $REV"
+        echo "revision=$REV" >> "$GITHUB_OUTPUT"
+
+    - name: Deploy
+      shell: bash
+      run: |
+        gcloud run deploy "${{ inputs.service }}" \
+          --image "${{ inputs.image }}" \
+          --region "${{ inputs.region }}" \
+          --platform managed \
+          --quiet
```

If we want to utilize the `deploy-cloudrun` github-action we can do somethink like this

 ```suggestion
     - id: 'deploy_green'
       name: Deploy new tagged-revision in green version
       uses: 'google-github-actions/deploy-cloudrun@v2'
       with:
         service: ${{ inputs.service }}
         image: ${{ inputs.image }}
         region: ${{ inputs.region }}
         tag_traffic: 'green=0' # this mean the new release will be having 0% traffic directed to it
     - id: 'health_check_green'
       name: Check if healthcheck path in green version returns healthy responses
       if: ${{ inputs.health_check_url }}
       shell: bash
       run: |
         for url in "${{ steps.deploy_green.outputs.url }}/${{ inputs.health_check_path }}"; do
           echo "Checking health of $url"
           RESULT=`curl -m 10 -s -o /dev/null -w "%{http_code}" "$url"`
           echo "Status code is $RESULT"
           if [ $RESULT -ge 300 ]; then
             echo "Health check failed"
             echo "error=true" >> "$GITHUB_OUTPUT"
             break
           fi
         done
```

---

@rot1024 updated the suggestion to supply the Host Header